lib/logproto/logproto-text-server.c

```diff
@@ -381,30 +203,37 @@ log_proto_text_server_fetch_from_buffer(LogProtoBufferedServer *s, const guchar
           || log_proto_buffered_server_is_input_closed(&self->super))
         {
           log_proto_text_server_yield_whole_buffer_as_message(self, state, buffer_start, buffer_bytes, msg, msg_len);
+          goto success;
         }
-      else
-        {
-          log_proto_text_server_split_buffer(self, state, buffer_start, buffer_bytes);
-          goto exit;
-        }
+
+      return FALSE;
     }
-  else if (!log_proto_text_server_extract(self, state, buffer_start, buffer_bytes, eol, msg, msg_len))
+
+  if (log_proto_text_server_extract(self, state, buffer_start, buffer_bytes, eol, msg, msg_len))
+    goto success;
+
+  if (log_proto_text_server_message_size_too_large(self, buffer_bytes))
     {
-      if (log_proto_text_server_message_size_too_large(self, buffer_bytes))
-        {
-          log_proto_text_server_yield_whole_buffer_as_message(self, state, buffer_start, buffer_bytes, msg, msg_len);
-        }
-      else
-        {
-          log_proto_text_server_split_buffer(self, state, buffer_start, buffer_bytes);
-          goto exit;
-        }
+      log_proto_text_server_yield_whole_buffer_as_message(self, state, buffer_start, buffer_bytes, msg, msg_len);
+      goto success;
     }
 
+  return FALSE;
+
+success:
   log_proto_text_server_remove_trailing_newline(msg, msg_len);
```

Comment:
looks great so far.

if you pushed this call back to the caller that would still simplify this function by removing all the goto statements here.

Reply:
I'd leave it as is if that's okay with you. I prefer `_fetch_msg_from_buffer()` to produce a "cleaned-up" message without the new lines.

Reply:
Sure.

Reply:
Thank you.